ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/jca/XATerminatorUnitTest.java

```diff
@@ -417,6 +423,153 @@ public void testConcurrentImport () throws Exception {
         assertEquals("some transaction import futures did not complete", completionCount.get(), TASK_COUNT);
     }
 
+    @Test
+    public void testRecovery() throws Exception {
+        Implementations.initialise();
+        XATerminatorImple xa = new XATerminatorImple();
+        Xid[] recover = xa.recover(XAResource.TMSTARTRSCAN);
+        int initialLength = recover == null ? 0 : recover.length;
```

Comment:
According to the javadoc the recover method will never return null. I know our implementation of it can return null but maybe we need to raise a JIRA to fix it to return an empty array instead.